### PR TITLE
Update gitignore

### PR DESCRIPTION
**Pull request by baer**
This PR updates the `.gitignore` file with a version that is auto-generated by www.gitignore.io. 

https://www.gitignore.io/api/osx,node,linux,eclipse,windows,intellij

Hopefully this will help keep custom ignores at a minimum and keep the fuss over things to ignore to a minimum.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,12 +1,259 @@
...
+# .idea/misc.xml
+# *.ipr
+
+# Sonarlint plugin
+.idea/sonarlint
+
+### Linux ###
 *~
-*.iml
-.*.haste_cache.*
```
Comment (baer) [inline .gitignore]: Note: This is the only line in this PR that is removed (and not just moved into another section). This appears to be a Jest specific cache file that does not apply to this project.

**Comment by leebyron** [comment]
I would prefer not track all files relevant to editors and other software in this project. Ideally our .gitignore should get smaller, not larger.

Any custom ignores related to software you use on your own machine should go in your [global .gitignore file](https://help.github.com/articles/ignoring-files/#create-a-global-gitignore)

**Comment by baer** [comment]
I was only trying to flesh out the things that had already been identified. Since you don't think editor specific things are appropriate here (I agree) I was going to suggest removing them but it looks like you did this yourself in master 👏  last night.

**Comment by wincent** [comment]
Good resolution to this! Thanks @baer for triggering the discussion and @leebyron for cleaning up the file. 👍

**Pull request closed without merging**